bundles/automation/org.eclipse.smarthome.automation.core/src/main/java/org/eclipse/smarthome/automation/core/Activator.java

```diff
@@ -71,7 +72,11 @@ public Object addingService(ServiceReference reference) {
                 StorageService storage = (StorageService) bc.getService(reference);
                 if (storage != null) {
                     final ManagedRuleProvider rp = new ManagedRuleProvider(storage);
-                    ruleRegistry = new RuleRegistryImpl(rm, rp);
+
+                    Storage storageDisabledRules = storage.getStorage("automation_rules_disabled",
+                            this.getClass().getClassLoader());
+
+                    ruleRegistry = new RuleRegistryImpl(rm, rp, storageDisabledRules);
```

Comment:
Yes I understand your comment, but at the moment the RuleRegistry is needed of ManagedRuleProvider which is an AbstractManagedProvider which can not exist without storage. 
May be we have to change impl. in following way:
1) RuleRegistry has to be created without listening of storage service, and create it without ManagedRuleProvider. 
2) ManagedRuleProvider has to be created only when storage service appears and it will be set by the addProvider method. 
3) additional enable/disable storage will be inject into RuleRegistry when storage service appears.

I prefer to create an additional issue for this, because at the moment the Rule engine does not work and this issue contains fix for it.

2)